### DISCUSSION TITLE
Dreaming to use the BTT Sensor with E3V2-UBL-BLTouch-15x15-v4.2.2-v1.3.5

### DISCUSSION THREAD

**Discussion by BlueWings172** [Jyers/Marlin#1239] [General Discussion]
Hello

I am using the E3V2-UBL-BLTouch-15x15-v4.2.2-v1.3.5 firmware and I have a BIGTREETECH Smart Filament Sensor that has been sitting on my desk for 2 months now because I don't know how to compile firmware. 

Can anyone offer any assistance ? My objective is to obtain a E3V2-UBL-BLTouch-15x15-v4.2.2-v1.3.5 firmware that can work with the sensor.

thanks !

**Comment by Cal-LF** [comment]
Have you thought about the https://www.patreon.com/jyers to get a custom FW? I'm thinking @Jyres can add the BTT smart sensor if it isn't already available.

**Reply by salymy10000** [reply]
#1248

**Reply by BlueWings172** [reply]
> Have you thought about the https://www.patreon.com/jyers to get a custom FW? I'm thinking @jyres can add the BTT smart sensor if it isn't already available.

That's a great idea.

I have just registered as a patreon to backup this project. Hopefully this helps getting the BTT to work with the E3V2-UBL-BLTouch-15x15-v4.2.2-v1.3.5 firmware.

**Reply by Cal-LF** [reply]
@BlueWings172 Reach out to @jyres if you have any questions. I hope you get the sensor working with your setup. I use the manual mesh 4X4 FW and the sensor work well. Although I have not had it trip other than by testing it.

**Comment by BlueWings172** [comment]
@Cal-LF  & @Jyers 

Thanks so much guys for your great help!

I did receive a link to a shared folder on dropbox so this is actually working !

**Reply by Jyers** [reply]
Thanks so much for deciding to support my work! I really hope you enjoy the auto-compile service 😄 

As always, feel free to reach out if you need any help at all!

**Comment by BlueWings172** [comment]
Hi Jacob

It is absolutely my pleasure to be backing very talented and innovative
developers like you.


I have tried the auto compile service and I really love the idea behind it.
That said I must be doing something wrong because I couldn't find the bin
file in the shared folder even after 24 hours.

Can you please let me know what I'm going wrong ?

Thanks so much.

Regards


Saleem

**Reply by Jyers** [reply]
So sorry you were having trouble. I checked my logs and it seems that the issue was the fact that you had a folder and a simplified config both named the same. Looks like you've removed the folder now, so you should get a binary soon. Sorry for the confusion, I'll have to add a catch for that situation to the script.

**Comment by BlueWings172** [comment]
I have removed the folder but that did not work. I also have renamed the
file to a short name without any special characters but that also did not
work.

Do you have any suggestions?

Thanks

**Reply by Jyers** [reply]
Sorry for all the confusion, seems like the issue may have been on my end. Either way it seems you now have a binary file 🥳

**Comment by BlueWings172** [comment]
Hey man

Thanks so much for your great support !

I do see a bin file in Dropbox!

I will test it and let you know the result.

Thanks again

**Comment by BlueWings172** [comment]
After ordering a crimping kit, figuring out the correct wire order and figuring out that you need to change the sensor setting to High, it worked.

That said, there are two issues:

1-After you are notified that the filament ran out and asked by Octoprint to change the filament, the sensor is ignored from that point on and if the filament runs out again during that print there will be no action or notification. In fact, what is surprising is that the checkbox for the filament sensor in the firmware becomes unchecked after the initial filament runout notification. Therefore, the sensor is automatically disabled and will not trigger in the next prints until the box is manually checked again which will be undone by the very next filament runout trigger. (yes, I have clicked on Store Settings when I enabled the sensor).

2- The filament runout notification is displayed on the LCD screen but you can only resume by clicking the message on Octoprint. Would be great if you can resume the print after changing the filament from the printer without having to go to the pc and resume from there.


Thanks for your help.
